model/dce-stdio.cc

```diff
@@ -342,27 +299,19 @@ int dce_fclose_unconditional (FILE *file)
   // Note: it is important here not to call the Current function here
   // because we need to be able to run this function even if there is no context.
   // For example, this is why we have no call to NS_LOG_FUNCTION (Current () ...);
-  struct my_IO_FILE_plus *fp = (struct my_IO_FILE_plus *)file;
-  static struct my_IO_jump_t vtable;
-  memcpy (&vtable, fp->vtable, sizeof(struct my_IO_jump_t));
-  vtable.__close = (void*)my_close_unconditional;
-  vtable.__write = (void*)my_write_unconditional;
-  vtable.__seek = (void*)my_seek_unconditional;
-  fp->vtable = &vtable;
-  fclose (file);
+  // TODO
+  //fclose(file);
+  // Maybee dce_close(fileno(file)) ?
   return 0;
 }
 int dce_fclose_onexec (FILE *file)
 {
   // Note: it is important here not to call the Current function here
   // because we need to be able to run this function even if there is no context.
   // For example, this is why we have no call to NS_LOG_FUNCTION (Current () ...);
-  struct my_IO_FILE_plus *fp = (struct my_IO_FILE_plus *)file;
-  static struct my_IO_jump_t vtable;
-  memcpy (&vtable, fp->vtable, sizeof(struct my_IO_jump_t));
-  vtable.__close = (void*)my_close_unconditional;
-  fp->vtable = &vtable;
-  fclose (file);
+  // TODO: fclose here might cause dce_write and will cause dce_close which calls Current()...
+  // but it works in the tests now.
+  fclose(file);
```

Comment:
The failure with the script that I mentioned in one of my [comments](https://github.com/direct-code-execution/ns-3-dce/pull/128#issuecomment-897072404), is because of this line itself. So, originally in our vtable mangling implementation, we overwrite only the closing operation with a ```my_close_unconditional``` which will not flush the stream, and is essentially a blank method, but we don't overwrite any other vtable callbacks. We then call fclose()  on it. But if we don't remap the close callback, and call ```fclose()``` it makes a call to dce_close() which will pop the fd from the openStreams stack, making it return back a -1 from the ```OPENED_FD_METHOD_ERR``` lookup when it runs ```dce_write``` . And so we don't get any output in the logs. Even avoiding a ```fclose()``` won't work in this case, as it will then fail for tests like dce-process-manager, because we won't be closing open FILE streams in ```dce_lclose_unconditional()``` as well, leading to a SIGSEGV when Libc flushed all streams.